gdal_subsetter/utilities.py

```diff
@@ -83,9 +84,50 @@ def get_files_from_unzipfiles(extract_dir: str, filetype: str,
     return ch_filelist
 
 
-def rename_file(input_filename: str, href: str) -> str:
-    """ Rename a given file to a name generated by the harmony-service-lib. """
+def rename_file(input_filename: str, stac_asset_href: str) -> str:
+    """ Rename a given file to a name determined for the input STAC Asset
+        by the harmony-service-lib Python library.
+
+        TODO: `generate_output_filename` should be called with appropriate
+              values for `variable_subset`, `is_regridded` and `is_subsetted`.
+              These kwargs allow the function to determine any required
+              suffices for the file, e.g., `<input_file>_subsetted.nc4`.
```

Comment:
I'm a little wary of implementing this right now - it would be a somewhat breaking change in that a lot of the output file names will change.

Reply:
I think it's mainly that it will break our regression tests. But I don't know if PO.DAAC are relying on the names being as they currently are when generating browse imagery (if they are still using HGA for that).